src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1261,63 +1268,84 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
+            {
+                callbackHandle.onSuccess(authResult);
+                return authResult;
+            }
+        } 
+        
+        // refresh token does not exist or refresh token request failed to give back the token. 
+        // If it's non-silent request, will try to acquire token interactively. 
+        // if it's silent request, will return the AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED.
+        if (refreshItem == null || authResult == null || (authResult != null && StringExtensions.IsNullOrBlank(authResult.getAccessToken()))) 
+        {
+            Logger.v(TAG, "Refresh token is not available or refresh token request failed to return token.");
             if (!request.isSilent() && callbackHandle.callback != null
-                    && (activity != null || useDialog)) {
-            	//Check if there is network connection
-                if (!mConnectionService.isConnectionAvailable()) {
-                    AuthenticationException exc = new AuthenticationException(
-                            ADALError.DEVICE_CONNECTION_IS_NOT_AVAILABLE,
-                            "Connection is not available to request token");
-                    Logger.w(TAG, "Connection is not available to request token", request.getLogInfo(),
-                            ADALError.DEVICE_CONNECTION_IS_NOT_AVAILABLE);
-                    callbackHandle.onError(exc);
-                    return null;
-                }
-                
-                // start activity if other options are not available
-                // delegate map is used to remember callback if another
-                // instance of authenticationContext is created for config
-                // change or similar at client app.
-                mAuthorizationCallback = callbackHandle.callback;
-                request.setRequestId(callbackHandle.callback.hashCode());
-                Logger.v(TAG, "Starting Authentication Activity with callback:"
-                        + callbackHandle.callback.hashCode());
-                putWaitingRequest(callbackHandle.callback.hashCode(),
-                        new AuthenticationRequestState(callbackHandle.callback.hashCode(), request,
-                                callbackHandle.callback));
-
-                if (useDialog) {
-                    AuthenticationDialog dialog = new AuthenticationDialog(mHandler, mContext,
-                            this, request);
-                    dialog.show();
-                } else {
-                    // onActivityResult will receive the response
-                    if (!startAuthenticationActivity(activity, request)) {
-                        callbackHandle.onError(new AuthenticationException(
-                                ADALError.DEVELOPER_ACTIVITY_IS_NOT_RESOLVED));
-                    }
-                }
-            } else {
-
+                    && (activity != null || useDialog)) 
+            {
```

Comment:
@RandalliLama ,the problem isn't really code convention here, but that this convention change and refactoring that increases PR...And I've got merge failure once just because of this.